File: python/src/easy/9/solution.py
from src.helper import Fields, tester

test_data = [
    {Fields.args: (121,), Fields.expd: True},
    {Fields.args: (-121,), Fields.expd: False},
    {Fields.args: (10,), Fields.expd: False},
]


class Solution:
    def isPalindrome(self, x: int) -> bool:  # noqa: N802
        if x <= 10:
            return False
        numbers = [num for num in str(x)]
        r = 0
        l = len(numbers) - 1
        while l > r:
            if numbers[r] == numbers[l]:
                r += 1
                l -= 1
            else:
                return False
        return True


if __name__ == "__main__":
    solution = Solution()
    tester(func=solution.isPalindrome, test_data=test_data)

File: go/easy/9/solution.go
package main

// func isPalindrome(x int) bool {
// 	if x < 0 {
// 		return false
// 	}
// 	numbers := []int{}
// 	for x > 0 {
// 		numbers = append(numbers, x%10)
// 		x /= 10
// 	}
// 	r := 0
// 	l := len(numbers) - 1
// 	for l > r {
// 		if numbers[r] == numbers[l] {
// 			r += 1
// 			l -= 1
// 		} else {
// 			return false
// 		}
// 	}
// 	return true
// }

func isPalindrome(x int) bool {
	if x < 0 {
		return false
	}
	origin := x
	rev := 0
	for x > 0 {
		rev = rev*10 + x%10
		x /= 10
	}
	return rev == origin
}

//12345
// 0*10 + 5 = 5 // 1234
// 5*10 + 4 = 54 // 123
// 54*10 + 3 = 543 // 12
// 543*10 + 2 = 5432 // 1
// 5432*10 + 1 = 54321 // 0
